controllers/goharbor/harbor/exporter.go

```diff
@@ -0,0 +1,112 @@
+package harbor
+
+import (
+	"context"
+	"net/url"
+
+	goharborv1 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha3"
+	harbormetav1 "github.com/goharbor/harbor-operator/apis/meta/v1alpha1"
+	"github.com/goharbor/harbor-operator/controllers"
+	"github.com/goharbor/harbor-operator/pkg/graph"
+	"github.com/goharbor/harbor-operator/pkg/version"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Exporter graph.Resource
+
+func (r *Reconciler) AddExporter(ctx context.Context, harbor *goharborv1.Harbor, core Core, certificate ExporterInternalCertificate) (Exporter, error) {
+	if harbor.Spec.Exporter == nil {
+		return nil, nil
+	}
+
+	exporter, err := r.GetExporter(ctx, harbor)
+	if err != nil {
+		return nil, errors.Wrap(err, "get")
+	}
+
+	exporterRes, err := r.AddBasicResource(ctx, exporter, core, certificate)
+
+	return Exporter(exporterRes), errors.Wrap(err, "add")
+}
+
+func (r *Reconciler) AddExporterConfigurations(ctx context.Context, harbor *goharborv1.Harbor, tlsIssuer InternalTLSIssuer) (ExporterInternalCertificate, error) {
+	if harbor.Spec.Exporter == nil {
+		return nil, nil
+	}
+
+	certificate, err := r.AddExporterInternalCertificate(ctx, harbor, tlsIssuer)
+	if err != nil {
+		return nil, errors.Wrap(err, "certificate")
+	}
+
+	return certificate, nil
+}
+
+type ExporterInternalCertificate graph.Resource
+
+func (r *Reconciler) AddExporterInternalCertificate(ctx context.Context, harbor *goharborv1.Harbor, tlsIssuer InternalTLSIssuer) (ExporterInternalCertificate, error) {
+	cert, err := r.GetInternalTLSCertificate(ctx, harbor, harbormetav1.ExporterTLS)
+	if err != nil {
+		return nil, errors.Wrap(err, "get")
+	}
+
+	certRes, err := r.Controller.AddCertificateToManage(ctx, cert, tlsIssuer)
+	if err != nil {
+		return nil, errors.Wrap(err, "add")
+	}
+
+	return ExporterInternalCertificate(certRes), nil
+}
+
+func (r *Reconciler) GetExporter(ctx context.Context, harbor *goharborv1.Harbor) (*goharborv1.Exporter, error) {
+	name := r.NormalizeName(ctx, harbor.GetName())
+	namespace := harbor.GetNamespace()
+
+	tls := harbor.Spec.InternalTLS.GetComponentTLSSpec(r.GetInternalTLSCertificateSecretName(ctx, harbor, harbormetav1.ExporterTLS))
+
+	host := r.NormalizeName(ctx, harbor.GetName(), controllers.Core.String())
+	if harbor.Spec.InternalTLS.IsEnabled() {
+		host += ":443"
+	} else {
+		host += ":80"
+	}
```

Comment:
Is it really useful? Scheme is already specified line 76.

```suggestion
```

Reply:
This is the same logic to get the core URL in the jobservice. I will extract these lines to be a method, jobservice and exporter will use this new method to get the URL of the core.